pkg/kubelet/kubelet.go

```diff
@@ -995,6 +995,30 @@ func (kl *Kubelet) podFieldSelectorRuntimeValue(fs *api.ObjectFieldSelector, pod
 	return fieldpath.ExtractFieldPathAsString(pod, internalFieldPath)
 }
 
+func (kl *Kubelet) lookupDnsService() (string, string, error) {
+	// Lookup the cluster DNS service and returns its info
+	if kl.serviceLister == nil {
+		return "", "", nil
+	}
+	services, err := kl.serviceLister.List()
```

Comment:
Here's how I understand it: This code pulls from a cached service listing that is kept up to date in the kubelet. In fact this is the same service listing that drives the container environment variables. See ```Kubelet.getServiceEnvVarMap()```

 * ```serviceLister``` is set to a ```StoreToServiceLister``` based on ```serviceStore```
 * ```serviceStore``` is the result of ```cache.NewStore``` and is configured to list/watch services in the ```NewMainKubelet``` function.
 * a ```StoreToServiceLister``` is just a way of making a store enumerable.

Have I got this right? If so, then this is light weight and should work fine.